utils/state.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"encoding/json"
+	"os"
+)
+
+// StateManager - An interface to use when stubbing
+// the filesystem operations. Implements the method signatures
+// from ioutil. see structs.go - IOStateManager for a reference
+// implementation
+type StateManager interface {
+	ReadFile() ([]byte, error)
+	WriteFile(data []byte, perm os.FileMode) error
+}
```

Comment:
perhaps neither here nor there, since this is just being used to make testing easier, but if we anticipated having other storage options for state in the future, I think this could be refactored to look a bit more like 

```go
type StateManager interface {
    ReadState() ([]Secret, error)
    WriteState(data []Secret) error
}
```

This abstracts it entirely away from the filesystem. At that point, you wouldn't really need the `ReadState` and `WriteState` functions below, as the IO and data marshal/unmarshal would be implementation dependent (e.g. it would live in IOStateManager)